src/components/network-map-tab.js

```diff
@@ -561,15 +561,30 @@ export const NetworkMapTab = ({
 
         dispatch(resetMapReloaded());
 
+        const currentNodeAtReloadCalling = currentNodeRef.current;
+
         return mapEquipments
             .reloadImpactedSubstationsEquipments(
                 studyUuid,
                 currentNode,
-                updatedSubstationsToSend,
-                setUpdatedLines
+                updatedSubstationsToSend
             )
-            .finally(() => {
-                setWaitingLoadData(false);
+            .then((results) => {
+                if (currentNodeAtReloadCalling === currentNodeRef.current) {
+                    const isFullReload = updatedSubstationsToSend
+                        ? false
+                        : true;
+                    mapEquipments.updateSubstations(
+                        mapEquipments.checkAndGetValues(results[0]),
+                        isFullReload
+                    );
+                    mapEquipments.updateLines(
+                        mapEquipments.checkAndGetValues(results[1]),
+                        isFullReload
+                    );
+                    setUpdatedLines(results[1]);
+                    setWaitingLoadData(false);
```

Comment:
Why did you remove the `.finally(() => {setWaitingLoadData(false);}` and put                     `setWaitingLoadData(false);` in the `.then` ?